File: Java/src/main/java/EnvironmentDatabase.java
import java.util.ArrayList;

public class EnvironmentDatabase {
    private ArrayList<EnvironmentData> historicalData;

    public EnvironmentDatabase(){
        historicalData = new ArrayList<EnvironmentData>();
    }

    public void addEnvironmentData(EnvironmentData envData){
        historicalData.add(envData);
    }

    public EnvironmentData getHistoricalEnvironmentData(int targetTime){
        return historicalData.get(targetTime);
    }
}

File: Python/knowledge_layer/EnvironmentDatabase.py
from typing import List

from specification.EnvironmentData import EnvironmentData


class EnvironmentDatabase:

    def __init__(self):
        self._historicalData: List[EnvironmentData] = []

    def addEnvironmentData(self, envData: EnvironmentData):
        self._historicalData.append(envData)

    def getHistoricalEnvironmentData(self) -> List[EnvironmentData]:
        return self._historicalData
